### PR TITLE
add support for email credentials

### LINKED ISSUE #230: [BUG] Support Email credentials from Alerting
**Describe the bug**
Notification does not support retrieving email username and password from keystore using the Alerting plugin's path that is shown [here](https://github.com/opensearch-project/alerting/blob/main/alerting/src/main/kotlin/org/opensearch/alerting/settings/DestinationSettings.kt#L53-L65).

**To Reproduce**
N/A

**Expected behavior**
Support retrieving credentials from there as a fallback

**Plugins**
N/A

**Screenshots**
N/A

**Host/Environment (please complete the following information):**
N/A

**Additional context**

### PR DESCRIPTION
**Pull request by zhongnansu**
### Description
- Add email username/password as plugin settings
- In setting, initialize a mapping of `accontName` to `DestinationSettings(username and password)`
- In `EmailClient` apply a logic to find the credentials and use it to configure the `mail client`, when method is `SSL` or `TLS`
- Add reload API to reload username and passoword, this is useful when cluster is up and running, and user wants to update crediantals using `opensearch-keystore`

### Issues Resolved
#230 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: notifications/spi/src/main/kotlin/org/opensearch/notifications/spi/setting/PluginSettings.kt [hunk 1]
```diff
@@ -293,6 +313,20 @@ internal object PluginSettings {
         NodeScope, Dynamic
     )
 
+    private val EMAIL_USERNAME: Setting.AffixSetting<SecureString> = Setting.affixKeySetting(
+        EMAIL_DESTINATION_SETTING_PREFIX,
+        "username",
+        // Needed to coerce lambda to Function type for some reason to avoid argument mismatch compile error
+        { key: String -> SecureSetting.secureString(key, null) }
```
Comment (joshuali925) [inline notifications/spi/src/main/kotlin/org/opensearch/notifications/spi/setting/PluginSettings.kt:320]: i checked affixKeySetting function type, lambda seems expected here, is the comment needed?

**Review by joshuali925 — APPROVED** [review]
seems AK's #257 might have some conflicts with this one

File: notifications/spi/src/main/kotlin/org/opensearch/notifications/spi/NotificationSpiPlugin.kt [hunk 2]
```diff
@@ -69,4 +72,8 @@ internal class NotificationSpiPlugin : Plugin() {
         PluginSettings.addSettingsUpdateConsumer(clusterService)
         return listOf()
     }
+
+    override fun reload(settings: Settings) {
```
[the remark this answers is not in the document]
Comment (zhongnansu) [reply notifications/spi/src/main/kotlin/org/opensearch/notifications/spi/NotificationSpiPlugin.kt:76]: https://opensearch.org/docs/monitoring-plugins/alerting/monitors/#authenticate-sender-account
https://www.elastic.co/guide/en/elasticsearch/reference/7.x/cluster-nodes-reload-secure-settings.html
when useing reload API

File: notifications/spi/src/main/kotlin/org/opensearch/notifications/spi/setting/PluginSettings.kt (hunk 1 above)
Comment (zhongnansu) [reply notifications/spi/src/main/kotlin/org/opensearch/notifications/spi/setting/PluginSettings.kt:320]: will remove

File: notifications/spi/src/main/kotlin/org/opensearch/notifications/spi/setting/PluginSettings.kt [hunk 3]
```diff
@@ -141,6 +149,11 @@ internal object PluginSettings {
      */
     private const val DEFAULT_TOOLTIP_SUPPORT = false
 
+    /**
+     * Default disable tooltip support
```
[the remark this answers is not in the document]
Comment (zhongnansu) [reply notifications/spi/src/main/kotlin/org/opensearch/notifications/spi/setting/PluginSettings.kt]: will change

**Comment by zhongnansu** [comment]
> seems AK's #257 might have some conflicts with this one

resolved the merge conflicts

**Review by davidcui1225 — APPROVED** [review]

**Review by joshuali925 — APPROVED** [review]

**Pull request merged**